{"error":"Illegal arguments: number, number","level":"error","message":"Registeration error :","timestamp":"2025-06-20T13:41:14.413Z"}
{"error":"Illegal arguments: number, number","level":"error","message":"Unhandled error :","stack":"Error: Illegal arguments: number, number\n    at _async (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\bcryptjs\\umd\\index.js:233:15)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\bcryptjs\\umd\\index.js:245:11\n    at new Promise (<anonymous>)\n    at Object.hash (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\bcryptjs\\umd\\index.js:244:16)\n    at Register (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:34:41)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-20T13:41:14.445Z"}
{"error":"No token provided","level":"error","message":"Authentication error :","timestamp":"2025-06-23T22:29:23.436Z"}
{"error":"No token provided","level":"error","message":"Unhandled error :","stack":"Error: No token provided\n    at authenticate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Middlewares\\auth.middleware.ts:53:10)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at Route.dispatch (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:117:3)\n    at handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:435:11)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:295:15\n    at processParams (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:582:12)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:291:5)\n    at Function.handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:186:3)","timestamp":"2025-06-23T22:29:23.643Z"}
{"error":"invalid token","level":"error","message":"Authentication error :","timestamp":"2025-06-23T22:29:54.925Z"}
{"error":"invalid token","level":"error","message":"Unhandled error :","stack":"Error: invalid token\n    at authenticate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Middlewares\\auth.middleware.ts:53:10)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at Route.dispatch (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:117:3)\n    at handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:435:11)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:295:15\n    at processParams (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:582:12)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:291:5)\n    at Function.handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:186:3)","timestamp":"2025-06-23T22:29:55.034Z"}
{"error":"notNull Violation: Budget.createdAat cannot be null","level":"error","message":"Error creating budget :","timestamp":"2025-06-23T22:30:16.086Z"}
{"error":"notNull Violation: Budget.createdAat cannot be null","level":"error","message":"Unhandled error :","stack":"SequelizeValidationError: notNull Violation: Budget.createdAat cannot be null\n    at InstanceValidator._validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:78:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async InstanceValidator._validateAndRunHooks (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:111:7)\n    at async InstanceValidator.validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:93:12)\n    at async Budget.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4077:7)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:46:20)","timestamp":"2025-06-23T22:30:16.203Z"}
{"error":"Email already registered","level":"error","message":"Registeration error :","timestamp":"2025-06-23T22:31:48.512Z"}
{"error":"Email already registered","level":"error","message":"Unhandled error :","stack":"Error: Email already registered\n    at Register (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:31:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T22:31:48.519Z"}
{"error":"Email already registered","level":"error","message":"Registeration error :","timestamp":"2025-06-23T22:32:11.573Z"}
{"error":"Email already registered","level":"error","message":"Unhandled error :","stack":"Error: Email already registered\n    at Register (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:31:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T22:32:11.580Z"}
{"error":"User not found","level":"error","message":"Login error :","timestamp":"2025-06-23T22:36:51.992Z"}
{"error":"User not found","level":"error","message":"Unhandled error :","stack":"Error: User not found\n    at Login (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:98:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T22:36:52.021Z"}
{"error":"notNull Violation: Budget.createdAat cannot be null","level":"error","message":"Error creating budget :","timestamp":"2025-06-23T22:40:53.630Z"}
{"error":"notNull Violation: Budget.createdAat cannot be null","level":"error","message":"Unhandled error :","stack":"SequelizeValidationError: notNull Violation: Budget.createdAat cannot be null\n    at InstanceValidator._validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:78:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async InstanceValidator._validateAndRunHooks (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:111:7)\n    at async InstanceValidator.validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:93:12)\n    at async Budget.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4077:7)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:46:20)","timestamp":"2025-06-23T22:40:53.753Z"}
{"error":"Missing required fields","level":"error","message":"Registeration error :","timestamp":"2025-06-23T22:52:51.515Z"}
{"error":"Missing required fields","level":"error","message":"Unhandled error :","stack":"Error: Missing required fields\n    at Register (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:23:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at Route.dispatch (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:117:3)\n    at handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:435:11)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:295:15\n    at processParams (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:582:12)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:291:5)\n    at Function.handle (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\index.js:186:3)","timestamp":"2025-06-23T22:52:51.569Z"}
{"error":"Budget already exists for this groceries, 500, 6, 2025","level":"error","message":"Error creating budget :","timestamp":"2025-06-23T22:54:21.946Z"}
{"error":"Budget already exists for this groceries, 500, 6, 2025","level":"error","message":"Unhandled error :","stack":"Error: Budget already exists for this groceries, 500, 6, 2025\n    at createBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:40:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T22:54:21.996Z"}
{"error":"Validation failed","level":"error","message":"Error creating budget :","timestamp":"2025-06-23T22:54:41.064Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at createBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:23:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T22:54:41.093Z"}
{"error":"notNull Violation: Transaction.createdAat cannot be null","level":"error","message":"Failed to create transaction :","timestamp":"2025-06-23T23:07:23.893Z"}
{"error":"notNull Violation: Transaction.createdAat cannot be null","level":"error","message":"Unhandled error :","stack":"SequelizeValidationError: notNull Violation: Transaction.createdAat cannot be null\n    at InstanceValidator._validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:78:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async InstanceValidator._validateAndRunHooks (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:111:7)\n    at async InstanceValidator.validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:93:12)\n    at async Transaction.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4077:7)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createTransaction (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\transaction.controller.ts:32:25)","timestamp":"2025-06-23T23:07:24.075Z"}
{"error":"Unexpected token '\"', \"\"\"\" is not valid JSON","level":"error","message":"Unhandled error :","stack":"SyntaxError: Unexpected token '\"', \"\"\"\" is not valid JSON\n    at JSON.parse (<anonymous>)\n    at createStrictSyntaxError (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:156:10)\n    at parse (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:71:15)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\read.js:123:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:211:14)\n    at invokeCallback (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at IncomingMessage.emit (node:domain:489:12)","timestamp":"2025-06-23T23:12:14.231Z"}
{"error":"Validation failed","level":"error","message":"Update budget failed :","timestamp":"2025-06-23T23:17:03.357Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at updateBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:195:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T23:17:03.405Z"}
{"error":"Budget not found","level":"error","message":"Get budget failed :","timestamp":"2025-06-23T23:18:34.358Z"}
{"error":"Budget not found","level":"error","message":"Unhandled error :","stack":"Error: Budget not found\n    at getBudget (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\budget.controller.ts:157:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T23:18:34.361Z"}
{"error":"User not found","level":"error","message":"Authentication error :","timestamp":"2025-06-23T23:26:23.457Z"}
{"error":"User not found","level":"error","message":"Unhandled error :","stack":"Error: User not found\n    at authenticate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Middlewares\\auth.middleware.ts:53:10)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T23:26:23.555Z"}
{"error":"User not found","level":"error","message":"Authentication error :","timestamp":"2025-06-23T23:26:49.081Z"}
{"error":"User not found","level":"error","message":"Unhandled error :","stack":"Error: User not found\n    at authenticate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Middlewares\\auth.middleware.ts:53:10)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-23T23:26:49.095Z"}
{"error":"Missing credentials for \"PLAIN\"","level":"error","message":"Email sending failed :","timestamp":"2025-06-23T23:30:31.957Z"}
{"error":"Missing credentials for \"PLAIN\"","level":"error","message":"Budget notifications error :","timestamp":"2025-06-23T23:30:31.998Z"}
{"error":"relation \"Goal\" does not exist","level":"error","message":"PostgreSQL connection error :","timestamp":"2025-06-26T20:43:53.009Z"}
{"error":"relation \"User\" does not exist","level":"error","message":"PostgreSQL connection error :","timestamp":"2025-06-26T21:17:25.511Z"}
{"error":"Validation failed","level":"error","message":"Create goal failed :","timestamp":"2025-06-26T21:24:46.929Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at createGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:22:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-26T21:24:47.073Z"}
{"error":"notNull Violation: Goal.currentAmount cannot be null","level":"error","message":"Create goal failed :","timestamp":"2025-06-27T18:21:28.752Z"}
{"error":"notNull Violation: Goal.currentAmount cannot be null","level":"error","message":"Unhandled error :","stack":"SequelizeValidationError: notNull Violation: Goal.currentAmount cannot be null\n    at InstanceValidator._validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:78:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async InstanceValidator._validateAndRunHooks (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:111:7)\n    at async InstanceValidator.validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:93:12)\n    at async Goal.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4077:7)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:28:18)","timestamp":"2025-06-27T18:21:28.914Z"}
{"error":"notNull Violation: Goal.currentAmount cannot be null","level":"error","message":"Create goal failed :","timestamp":"2025-06-27T18:22:17.646Z"}
{"error":"notNull Violation: Goal.currentAmount cannot be null","level":"error","message":"Unhandled error :","stack":"SequelizeValidationError: notNull Violation: Goal.currentAmount cannot be null\n    at InstanceValidator._validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:78:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async InstanceValidator._validateAndRunHooks (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:111:7)\n    at async InstanceValidator.validate (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\instance-validator.js:93:12)\n    at async Goal.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4077:7)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:28:18)","timestamp":"2025-06-27T18:22:17.659Z"}
{"error":"Validation failed","level":"error","message":"Create goal failed :","timestamp":"2025-06-27T18:33:11.642Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at createGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:22:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-27T18:33:11.694Z"}
{"error":"Email already registered","level":"error","message":"Registeration error :","timestamp":"2025-06-27T20:39:36.906Z"}
{"error":"Email already registered","level":"error","message":"Unhandled error :","stack":"Error: Email already registered\n    at Register (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\auth.controller.ts:31:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-27T20:39:36.958Z"}
{"error":"Validation failed","level":"error","message":"Create goal failed :","timestamp":"2025-06-27T20:41:18.007Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at createGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:22:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-27T20:41:18.063Z"}
{"error":"Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).","level":"error","message":"Get goal failed :","timestamp":"2025-06-27T20:48:12.462Z"}
{"error":"Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).","level":"error","message":"Unhandled error :","stack":"SequelizeEagerLoadingError: Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).\n    at Function._getIncludedAssociation (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:751:15)\n    at Function._validateIncludedElement (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:646:53)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:542:37\n    at Array.map (<anonymous>)\n    at Function._validateIncludedElements (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:537:39)\n    at Function.findAll (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:1794:12)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async Function.findOne (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:1998:12)\n    at async getGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:111:18)","timestamp":"2025-06-27T20:48:12.551Z"}
{"error":"Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).","level":"error","message":"Get goal failed :","timestamp":"2025-06-27T20:49:18.288Z"}
{"error":"Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).","level":"error","message":"Unhandled error :","stack":"SequelizeEagerLoadingError: Transaction is associated to Goal using an alias. You've included an alias (transaction), but it does not match the alias(es) defined in your association (transactions).\n    at Function._getIncludedAssociation (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:751:15)\n    at Function._validateIncludedElement (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:646:53)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:542:37\n    at Array.map (<anonymous>)\n    at Function._validateIncludedElements (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:537:39)\n    at Function.findAll (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:1794:12)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async Function.findOne (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:1998:12)\n    at async getGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:111:18)","timestamp":"2025-06-27T20:49:18.298Z"}
{"error":"Unexpected token '\"', \"\"\\r\\n{\\r\\n\"... is not valid JSON","level":"error","message":"Unhandled error :","stack":"SyntaxError: Unexpected token '\"', \"\"\\r\\n{\\r\\n\"... is not valid JSON\n    at JSON.parse (<anonymous>)\n    at createStrictSyntaxError (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:156:10)\n    at parse (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:71:15)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\read.js:123:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:211:14)\n    at invokeCallback (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at IncomingMessage.emit (node:domain:489:12)","timestamp":"2025-06-27T20:59:18.715Z"}
{"error":"status is not defined","level":"error","message":"Update goal failed :","timestamp":"2025-06-27T20:59:49.320Z"}
{"error":"status is not defined","level":"error","message":"Unhandled error :","stack":"ReferenceError: status is not defined\n    at updateGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:181:15)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-27T20:59:49.349Z"}
{"error":"status is not defined","level":"error","message":"Update goal failed :","timestamp":"2025-06-27T21:01:31.628Z"}
{"error":"status is not defined","level":"error","message":"Unhandled error :","stack":"ReferenceError: status is not defined\n    at updateGoal (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\goal.controller.ts:181:15)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-27T21:01:31.637Z"}
{"error":"insert or update on table \"Transactions\" violates foreign key constraint \"Transactions_goalId_fkey\"","level":"error","message":"Failed to create transaction :","timestamp":"2025-06-27T21:11:19.239Z"}
{"error":"insert or update on table \"Transactions\" violates foreign key constraint \"Transactions_goalId_fkey\"","level":"error","message":"Unhandled error :","stack":"Error\n    at Query.run (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\dialects\\postgres\\query.js:76:25)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\sequelize.js:650:28\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async PostgresQueryInterface.insert (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\dialects\\abstract\\query-interface.js:795:21)\n    at async Transaction.save (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:4154:35)\n    at async Function.create (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\sequelize\\src\\model.js:2305:12)\n    at async createTransaction (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\transaction.controller.ts:53:25)","timestamp":"2025-06-27T21:11:19.256Z"}
{"error":"Unexpected token '\"', \"\"{\\r\\n   \\\"... is not valid JSON","level":"error","message":"Unhandled error :","stack":"SyntaxError: Unexpected token '\"', \"\"{\\r\\n   \\\"... is not valid JSON\n    at JSON.parse (<anonymous>)\n    at createStrictSyntaxError (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:156:10)\n    at parse (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:71:15)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\read.js:123:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:211:14)\n    at invokeCallback (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at IncomingMessage.emit (node:domain:489:12)","timestamp":"2025-06-27T21:15:30.492Z"}
{"error":"Validation failed","level":"error","message":"Update subcription error :","timestamp":"2025-06-30T11:32:07.279Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at updateSubscription (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\subscription.controller.ts:153:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-30T11:32:07.530Z"}
{"error":"Validation failed","level":"error","message":"Update subcription error :","timestamp":"2025-06-30T11:34:59.995Z"}
{"error":"Validation failed","level":"error","message":"Unhandled error :","stack":"Error: Validation failed\n    at updateSubscription (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\src\\Controllers\\subscription.controller.ts:153:13)\n    at Layer.handleRequest (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\layer.js:152:17)\n    at next (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\router\\lib\\route.js:157:13)\n    at middleware (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\express-validator\\lib\\middlewares\\check.js:16:13)\n    at processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-06-30T11:35:00.095Z"}
{"error":"Unexpected token '\"', \"\"{\\r\\n    \"... is not valid JSON","level":"error","message":"Unhandled error:","stack":"SyntaxError: Unexpected token '\"', \"\"{\\r\\n    \"... is not valid JSON\n    at JSON.parse (<anonymous>)\n    at createStrictSyntaxError (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:156:10)\n    at parse (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\types\\json.js:71:15)\n    at C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\body-parser\\lib\\read.js:123:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:211:14)\n    at invokeCallback (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Viktorrrr!\\Desktop\\Personal_Finance_Manager\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at IncomingMessage.emit (node:domain:489:12)","timestamp":"2025-07-04T03:15:55.300Z"}
